posthog/performance/sql.py

```diff
@@ -0,0 +1,218 @@
+"""https://developer.mozilla.org/en-US/docs/Web/API/PerformanceEntry"""
+from posthog import settings
+from posthog.clickhouse.kafka_engine import STORAGE_POLICY, kafka_engine
+from posthog.clickhouse.table_engines import Distributed, MergeTree, ReplicationScheme
+from posthog.kafka_client.topics import KAFKA_PERFORMANCE_EVENTS
+
+KAFKA_COLUMNS_WITH_NULLABLE_DATETIME = """
+, _timestamp Nullable(DateTime)
+, _offset UInt64
+, _partition UInt64
+"""
+
+# TODO
+# explode server timing from Resource events into their own columns
+
+"""
+# expected queries
+
+## get all performance events for a given team's session
+
+allows us to show performance events alongside other logs while viewing a session recording
+
+shard by session id so that when querying for all in session or pageview all data for the results are on the same shard
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND session_id = 'my-session-uuid'
+ORDER BY start_time
+
+## get all performance events for a given team's pageview
+
+allows us to show performance events in a waterfall chart for a given pageview
+
+SELECT * FROM performance_events
+WHERE team_id = 1
+AND session_id = 'my-session-uuid'
+AND pageview_id = 'my-page-view-uuid' -- sent by SDK
```

Comment:
Always do this:

AND timestamp > ROUGH_LOWER_BOUND and timestamp < ROUGH_UPPER_BOUND

- LOWER/UPPER bound can be large (e.g. 24h) - allows skipping a bunch of partitions
- Because it makes performance consistent as amount of _historical_ data increases.
- Alongside the ORDER BY / PARTITION BY change proposed below